### PR TITLE
API Update avatar

### PR DESCRIPTION
**Pull request by Dabkdn**
Update avatar of current user

#11

### CODE REVIEW

File: EducationManagement/Services/Implementations/UserService.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            string base64Decoded;
+
+            byte[] data = Convert.FromBase64String(token);
+
+            base64Decoded = ASCIIEncoding.ASCII.GetString(data);
+
+            string[] temp = base64Decoded.Split('@');
+
+            var userId = Int32.Parse(temp[0]);
```
Comment (nguyenvanman) [inline EducationManagement/Services/Implementations/UserService.cs]: you can extract above codes to a function for using in future

File: EducationManagement/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        /// <summary>
+        /// update user avatar
+        /// </summary>
+        [Route("api/user/update/avatar")]
+        [HttpPost]
```
Comment (nguyenvanman) [inline EducationManagement/Controllers/UserController.cs]: the method should be HttpPut for a update action

File: EducationManagement/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        /// <summary>
+        /// update user avatar
+        /// </summary>
+        [Route("api/user/update/avatar")]
```
Comment (nguyenvanman) [inline EducationManagement/Controllers/UserController.cs]: i think route can be "api/users/avatar"

File: EducationManagement/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        [HttpPost]
+        public IHttpActionResult UpdateAvatar([FromBody] string url)
+        {
+            if(url == null)
+            {
+                return BadRequest();
+            }
+
+            var token = Request.Headers.GetValues("Token");
```
Comment (nguyenvanman) [inline EducationManagement/Controllers/UserController.cs]: please check the token and return an Unauthorized response if null

Comment (nguyenvanman) [reply EducationManagement/Controllers/UserController.cs]:
```if (Request.Headers.Contains("Token") { 
} else {
}```

File: EducationManagement/Services/Implementations/UserService.cs (hunk 1 above)
Comment (Dabkdn) [reply EducationManagement/Services/Implementations/UserService.cs]: It's reasonable.

File: EducationManagement/Controllers/UserController.cs (hunk 4 above)
Comment (Dabkdn) [reply EducationManagement/Controllers/UserController.cs]: So, if that token not null and be a Wrong token, What will you do?

**Pull request merged**